pkg/events/events.go

```diff
@@ -0,0 +1,77 @@
+package events
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+)
+
+// Event message format "medik8s <operator shortname> <message>"
+const customFmt = "[%3s] %s"
+
+type EventRecorder interface {
+	// NormalEventf will record an event with type Normal and fixed message.
+	NormalEvent(object runtime.Object, reason, message string)
+	// NormalEventf will record an event with type Normal and formatted message.
+	NormalEventf(object runtime.Object, reason, messageFmt string, a ...interface{})
+	// WarningEventf will record an event with type Warning and fixed message.
+	WarningEvent(object runtime.Object, reason, message string)
+	// WarningEventf will record an event with type Warning and formatted message.
+	WarningEventf(object runtime.Object, reason, messageFmt string, a ...interface{})
+
+	// Event overrides the default event recorder to add the operator shortname to the message.
+	Event(object runtime.Object, eventtype, reason, message string)
+	// Eventf overrides the default event recorder to add the operator shortname to the message.
+	Eventf(object runtime.Object, eventtype, reason, messageFmt string, args ...interface{})
+	// AnnotatedEventf overrides the default event recorder to add the operator shortname to the message.
+	AnnotatedEventf(object runtime.Object, annotations map[string]string, eventtype, reason, messageFmt string, args ...interface{})
+}
+
+type eventRecorder struct {
+	record.EventRecorder
+	operator string
+}
+
+var _ EventRecorder = &eventRecorder{}
+
+// NewEventRecorder returns a new EventRecorder that adds the operator shortname to the message.
+func NewEventRecorder(recorder record.EventRecorder, operatorShortName string) EventRecorder {
+	return &eventRecorder{
+		EventRecorder: recorder,
+		operator:      operatorShortName,
+	}
+}
+
+func (e *eventRecorder) NormalEvent(object runtime.Object, reason, message string) {
+	e.Event(object, "Normal", reason, message)
+}
+
+func (e *eventRecorder) NormalEventf(object runtime.Object, reason, messageFmt string, a ...interface{}) {
+	msg := fmt.Sprintf(messageFmt, a...)
+	e.NormalEvent(object, reason, msg)
+}
+
+func (e *eventRecorder) WarningEvent(object runtime.Object, reason, message string) {
+	e.Event(object, "Warning", reason, message)
+}
+
+func (e *eventRecorder) WarningEventf(object runtime.Object, reason, messageFmt string, a ...interface{}) {
+	msg := fmt.Sprintf(messageFmt, a...)
+	e.WarningEvent(object, reason, msg)
+}
+
+func (e *eventRecorder) Event(object runtime.Object, eventtype, reason, message string) {
```

Comment:
replace `eventtype` with `eventType`

Reply:
I copy/pasted the signature from upstream, but I can change it :)

https://pkg.go.dev/k8s.io/client-go/tools/record#EventRecorder